tests/callbacks/test_early_stopping.py

```diff
@@ -294,3 +294,27 @@ def validation_epoch_end(self, outputs):
         (trainer.global_step, max(min_steps, by_early_stopping, by_min_epochs), step_freeze, min_steps, min_epochs)
 
     _logger.disabled = False
+
+
+def test_early_stopping_mode_options():
+    with pytest.raises(MisconfigurationException) as exc_info:
+        EarlyStopping(mode="unknown_option")
+    assert "unknown_option" in str(exc_info.value)
+
+    with pytest.warns(DeprecationWarning) as exc_info:
+        EarlyStopping()
+
+    with pytest.warns(DeprecationWarning):
+        EarlyStopping(mode="auto")
+
+
+def test_model_checkpoint_mode_options():
+    with pytest.raises(MisconfigurationException) as exc_info:
+        ModelCheckpoint(mode="unknown_option")
+    assert "unknown_option" in str(exc_info.value)
+
+    with pytest.warns(DeprecationWarning) as exc_info:
+        ModelCheckpoint()
+
+    with pytest.warns(DeprecationWarning):
+        ModelCheckpoint(mode="auto")
```

Comment:
pls add also a matching pattern, so we are clear what error it is...

Reply:
Done!